sql/updates/reverse-dev.sql

```diff
@@ -1,3 +1,22 @@
+DO $$
+DECLARE
+    remove_these text;
+BEGIN
+    IF EXISTS (
+        SELECT
+            string_agg(format('%I.%I', user_view_schema, user_view_name), ', ') INTO remove_these
+        FROM
+            _timescaledb_catalog.continuous_agg
+            JOIN _timescaledb_catalog.hypertable ht ON raw_hypertable_id = ht.id
+        WHERE
+            NOT _timescaledb_functions.has_invalidation_trigger (format('%I.%I', ht.schema_name, ht.table_name))) THEN
+    RAISE EXCEPTION 'cannot downgrade because there are continuous aggregates using WAL-based invalidation collection'
+        USING
+            ERRCODE = 'object_not_in_prerequisite_state',
+            DETAIL = format('Please remove these CAggs before downgrade: %s.', remove_these);
```

Comment:
Can't we create missing invalidation triggers in those CAgg instead of blocking downgrade and ask user to drop it? Or can we switch from wal-based to trigger-base using `ALTER MATERIALIZED VIEW`?

Reply:
Not easily. You can have unprocessed invalidations in the WAL that you first need to process before adding the trigger. This is outlined in #8590, but I would like to have a blocker right now to prevent invalidations from getting lost while working on this.

Reply:
... and it is not possible to migrate between `trigger` and `wal` using `ALTER MATERIALIZED VIEW` yet. This is what #8590 will do.